File: data-sharing/src/main/java/xyz/chener/zp/datasharing/service/impl/UserModuleServiceFallback.java
package xyz.chener.zp.datasharing.service.impl;

import com.github.pagehelper.PageInfo;
import org.springframework.stereotype.Component;
import xyz.chener.zp.datasharing.entity.thirdparty.OrgBase;
import xyz.chener.zp.datasharing.entity.thirdparty.UserBase;
import xyz.chener.zp.datasharing.service.UserModuleService;

import java.util.Collections;
import java.util.List;

/**
 * @Author: chenzp
 * @Date: 2023/01/13/08:50
 * @Email: chen@chener.xyz
 */


@Component("userModuleServiceFallback")
public class UserModuleServiceFallback implements UserModuleService {


    @Override
    public PageInfo<UserBase> getUserBaseInfo(UserBase userBase, Integer page, Integer size) {
        return new PageInfo<>(Collections.emptyList());
    }

    @Override
    public List<OrgBase> getUserOrgs(String username) {
        return Collections.emptyList();
    }
}

File: task/src/main/kotlin/xyz/chener/zp/task/service/impl/UserModuleServiceFallBack.kt
package xyz.chener.zp.task.service.impl

import com.github.pagehelper.PageInfo
import org.springframework.stereotype.Service
import xyz.chener.zp.task.entity.thirdparty.OrgBase
import xyz.chener.zp.task.entity.thirdparty.UserBase
import xyz.chener.zp.task.service.UserModuleService

@Service
class UserModuleServiceFallBack : UserModuleService {
    override fun getUserBaseInfo(userBase: UserBase?, page: Int?, size: Int?): PageInfo<UserBase?>? {
        return PageInfo<UserBase?>().also {
            it.total = 0
        }
    }

    override fun getUserOrgs(username: String?): List<OrgBase?>? {
        return emptyList()
    }

    override fun getUserBaseInfoByUserIds(userIds: List<Long?>?): List<UserBase> {
        return emptyList()
    }
}